profiler/internal/stackparse/stackparse.go

```diff
@@ -0,0 +1,341 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+// Package stackparse parses goroutines stack trace dumps as produced by
+// runtime.Stack(). The design goals are:
+//
+// 1. Safe: No panics should be thrown.
+// 2. Simple: Keep this pkg small and easy to modify.
+// 3. Forgiving: Favor producing partial results over no results, even if the
+// input data is different than expected.
+// 4. Efficient: Parse several hundred MB/s.
+package stackparse
+
+// Before implementing this pkg we experimented with panicparse, but decided to
+// roll out our own library because of gomodule issues [1], and because we
+// figured we can make a library that's 100x faster while 10x less code.
+//
+// [1] https://github.com/maruel/panicparse/issues/57
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type parserState int
+
+const (
+	stateHeader = iota
+	stateStackFunc
+	stateStackFile
+	stateCreatedBy
+	stateCreatedByFunc
+	stateCreatedByFile
+)
+
+// Parse parses a goroutines stack trace dump as produced by runtime.Stack().
+// The parser is forgiving and will continue parsing even when encountering
+// unexpected data. When this happens it will try to discard the entire
+// goroutine that encountered the problem and continue with the next one. It
+// will also return an *Error including an error for every goroutine that
+// couldn't be parsed. If all goroutines were parsed successfully, *Errors is
+// nil.
+func Parse(r io.Reader) ([]*Goroutine, *Errors) {
+	var (
+		sc      = bufio.NewScanner(r)
+		state   parserState
+		lineNum int
+		line    []byte
+
+		goroutines []*Goroutine
+		g          *Goroutine
+		f          *Frame
+
+		errs           = &Errors{}
```

Comment:
stackparse.Errors is just a struct around a slice; wouldn't have to be implemented as a pointer?

Reply:
Changed it to `[]error` - de44d72